slides/src/content/rxjs-observables.md

```diff
@@ -1,164 +1,191 @@
+});
+getData$.subscribe((data) => {
+  // 
+});
 ```
 
-Similarly to `fromEvent`, RxJS library provides [a number of different ways to create Observables](http://reactivex.io/rxjs/manual/overview.html#creation-operators) such as `fromPromise`.
+In this example, `map` is used because `get` is an observable of an object with a `json` method.  Usually, we want the return value of `json()`, not the response object!
 
 ---
 
-## Disposing (Cancelling) Observable Execution
-
-Say, for some reason, we want to the button to stop printing `Clicked!` to console after 3 seconds. 
+## What about Unsubscribing?
 
-We can achieve this by *disposing* or *cancelling* the Observable execution. This is done by *unsubscribing* from the executing Observable.
+* Obserables that finish, including `http.get` don't need to be unsubscribed.
+* If our observable keeps pushing values, the function we gave it when we subscribed will be called until we unsubscribe. We can stop this by unsubscribing from the observable.
+* The `subscribe()` method returns a *subscription*.
+* A subscription has an `unsubscribe()` method.
 
 ```js
 const clickStreamSubscription = clickStream.subscribe(
-  () => console.log('Click me!')
+  () => console.log('clicked!')
 );
 
 setTimeout(() => {
   clickStreamSubscription.unsubscribe(); 
 }, 3000);
 ```
 
-Notice that subscribing to the Observable returns a *Subscription* object. We can then use this Subscription object's 
-`unsubscribe` method to dispose the Observable stream.
+Notes:
+*concept*: subscription, unsubscribe method
 
 ---
 
-## Observable HTTP Events in Angular
+## Angular's AsyncPipe
 
-One of the most common uses for Observables in Angular is to handle HTTP events:
+* The `AsyncPipe` helps us write less by taking an observable and:
+  - subscribing to it
+  - returning the returned value so it can be placed in your template
+  - unsubscibing, when the value is no longer needed
 
-```js 
-http.get('http://jsonplaceholder.typicode.com/users/')
-  .map((response) => Observable.from(response.json()))
-  .subscribe((data) => {
-    data.forEach(user => console.log(user.name));
-  });
-```
+Example (`name$` is an observable):
+
+`<div> {{ name$ | async }} </div>`
+
++++
+
+## Quiz 4
+
+Angular's `AsyncPipe`
+  1. accepts an observable and binds the value to a template
+  2. gets data from a rest endpoint
+  3. makes async code look procedural
+  4. can't be chained with pure pipes
 
-* `.map((data) => new Observable(data.json()))` transforms the `response` (response data) into a new Observable stream that emits `response` in a JSON object form.
-* We can then work with to response data as a JSON object by subscribing to the newly transformed Observable stream.
 
 ---
 
-## Observable Operations
+## The Filter Operators
 
-Using *Observable operators* we can perform Array-like operations to filter and transform values as they are emitted by the Observable execution: 
+In addition to `map`, other the filter operator is available: 
```

Comment:
I'm going to rework this example.  The filter in this case is an array filter, not an observable filter, so it's really confusing.  Instead I'm going to to change it to
```javascript
searchString$
  .filter(searchString => searchString.length > 3)
  .subscribe(value = console.log(value));
```
to pick up from the example from earlier.